### PR TITLE
adds moon_getLatestSyncedBlock to moonbeam custom API methods

### PR DESCRIPTION
**Pull request by dawnkelly09**
### Description

- adds moon_getLatestSyncedBlock to moonbeam custom API methods
- includes curl example script

### Checklist

- [ x ] I have added a label to this PR 🏷️
- [ ] I have run my changes through Grammarly
- [ ] If this requires translations for the `moonbeam-docs-cn` repo, I have created a ticket for the translations in Jira
- [ ] If pages have been moved around, I have created an additional PR in `moonbeam-mkdocs` to update redirects
- [ ] If pages have been moved around, I have run the `move-pages.py` script to move the pages and update the image paths on the chinese repo
    - [ ] After the script has been run, I have created an additional PR in `moonbeam-docs-cn`
- [ ] If images have been added, I have run the `compress-images.py` script to compress the images.
- [ ] If variables (in variables.yml) need to be updated (such as a name change), I have updated the `moonbeam-docs-cn` repo to use the new variables
- [ ] If this page requires a disclaimer, I have added one

### Corresponding PRs

Please link to any corresponding PRs here.

### After Translation Requirements

- [ ] Will need to create PR in `moonbeam-docs` repo to remove images
- [ ] Will need to create PR in `moonbeam-docs` repo to remove variables
- [ ] Will need to create PR in `moonbeam-mkdocs` repo to add redirects for Chinese site
- [ ] No additional PRs are required after the translations are done

#### Items to be Updated

Please list any of the items that will need to be added or deleted after the translations are done here.

### CODE REVIEW

File: builders/ethereum/json-rpc/moonbeam-custom-api.md [hunk 1]
```diff
@@ -7,7 +7,12 @@ description: Discover Moonbeam's specialized API endpoints, featuring custom JSO
 
 ## Introduction {: #introduction }
 
-Moonbeam nodes include support for two custom JSON-RPC endpoints: `moon_isBlockFinalized` and `moon_isTxFinalized`. These endpoints provide valuable functionality for checking the finality of on-chain events.
+Moonbeam nodes include support for custom JSON-RPC endpoints:
```
Comment (themacexpert) [inline builders/ethereum/json-rpc/moonbeam-custom-api.md:10]:
```suggestion
Moonbeam nodes include support for custom JSON-RPC endpoints: 

```

Comment (themacexpert) [reply builders/ethereum/json-rpc/moonbeam-custom-api.md:10]: Screenshot 2024-08-07 at 9 21 56 AM
Currently looks like this but the extra line should fix it.

File: builders/ethereum/json-rpc/moonbeam-custom-api.md [hunk 2]
```diff
@@ -58,3 +63,30 @@ To begin exploring Moonbeam's custom JSON-RPC endpoints, you can try out the pro
...
+
+???+ function "moon_getLatestSyncedBlock"
+
+    Returns the latest synced block number.
+
+    === "Parameters"
+
+        None
+
+    === "Returns"
```
Comment (themacexpert) [inline builders/ethereum/json-rpc/moonbeam-custom-api.md:75]: Also want to mention the data type that is being returned, in this case, an integer

**Review by themacexpert — CHANGES_REQUESTED** [review]

**Review by themacexpert — APPROVED** [review]
LGTM!

**Comment by eshaben** [comment]
Thank you both!

**Pull request merged**